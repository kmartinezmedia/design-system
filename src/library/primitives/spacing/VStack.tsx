import React, { FunctionComponent } from 'react';

import { View, ViewProps } from './View';

export const VStack: FunctionComponent<ViewProps> = ({
  spacing = 4,
  ...props
}) => <View {...props} spacing={spacing} />;

export type VStackProps = ViewProps;
