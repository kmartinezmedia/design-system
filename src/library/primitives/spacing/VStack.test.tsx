import React from 'react';
import 'react-native';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { VStack } from './VStack';

describe('VStack', () => {
  it('default', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<VStack />);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
