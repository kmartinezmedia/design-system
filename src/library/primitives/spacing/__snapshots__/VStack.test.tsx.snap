// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VStack default 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "alignSelf": "auto",
        "flexDirection": "column",
        "flexGrow": undefined,
        "flexShrink": undefined,
        "flexWrap": "nowrap",
        "height": undefined,
        "justifyContent": "flex-start",
        "width": undefined,
      },
      Object {
        "padding": 16,
      },
      undefined,
      undefined,
      Object {},
    ]
  }
/>
`;
